bootstrap/postgres/configs.go

```diff
@@ -47,53 +48,57 @@ func NewConfigRepository(db postgres.Database, log *slog.Logger) bootstrap.Confi
 	return &configRepository{db: db, log: log}
 }
 
-func (cr configRepository) Save(ctx context.Context, cfg bootstrap.Config, chsConnIDs []string) (string, error) {
-	q := `INSERT INTO configs (magistrala_thing, owner, name, client_cert, client_key, ca_cert, magistrala_key, external_id, external_key, content, state)
-		  VALUES (:magistrala_thing, :owner, :name, :client_cert, :client_key, :ca_cert, :magistrala_key, :external_id, :external_key, :content, :state)`
+func (cr configRepository) Save(ctx context.Context, cfg bootstrap.Config, chsConnIDs []string) (thingID string, err error) {
+	q := `INSERT INTO configs (magistrala_thing, domain_id, name, client_cert, client_key, ca_cert, magistrala_key, external_id, external_key, content, state)
+		  VALUES (:magistrala_thing, :domain_id, :name, :client_cert, :client_key, :ca_cert, :magistrala_key, :external_id, :external_key, :content, :state)`
 
 	tx, err := cr.db.BeginTxx(ctx, nil)
 	if err != nil {
 		return "", errors.Wrap(repoerr.ErrCreateEntity, err)
 	}
 
 	dbcfg := toDBConfig(cfg)
+	defer func() {
+		if err != nil {
+			err = cr.rollback("Save method", err, tx)
+		} else {
+			if commitErr := tx.Commit(); commitErr != nil {
+				err = commitErr
+			}
+		}
+	}()
 
 	if _, err := tx.NamedExec(q, dbcfg); err != nil {
-		e := err
-		if pgErr, ok := err.(*pgconn.PgError); ok && pgErr.Code == pgerrcode.UniqueViolation {
-			e = repoerr.ErrConflict
+		switch pgErr := err.(type) {
+		case *pgconn.PgError:
+			if pgErr.Code == pgerrcode.UniqueViolation {
+				err = repoerr.ErrConflict
+			}
 		}
-
-		cr.rollback("Failed to insert a Config", tx)
-		return "", errors.Wrap(repoerr.ErrCreateEntity, e)
+		return "", err
 	}
 
-	if err := insertChannels(ctx, cfg.Owner, cfg.Channels, tx); err != nil {
-		cr.rollback("Failed to insert Channels", tx)
-		return "", errors.Wrap(errSaveChannels, err)
+	if err := insertChannels(cfg.DomainID, cfg.Channels, tx); err != nil {
+		err = errors.Wrap(errSaveChannels, err)
+		return "", err
 	}
```

Comment:
What do you think about below suggestion
```suggestion
	if err := insertChannels(cfg.DomainID, cfg.Channels, tx); err != nil {
		return "", err = errors.Wrap(errSaveChannels, err)
	}
```

Reply:
I think it is more compact